src/web/app/pages-instructor/instructor-session-base-page.component.ts

```diff
@@ -282,20 +287,60 @@ export abstract class InstructorSessionBasePageComponent {
     const filename: string =
         `${model.feedbackSession.courseId}_${model.feedbackSession.feedbackSessionName}_result.csv`;
     let blob: any;
+    let downloadAborted: boolean = false;
+    const out: string[] = [];
+
+    let numberOfQuestionsDownloaded: number = 0;
+    const modalContent: string =
+    'Downloading the results of your feedback session...';
+    const loadingModal: NgbModalRef = this.simpleModalService.openLoadingModal(
+        'Download Progress', SimpleModalType.LOAD, modalContent);
+    loadingModal.result.then(() => {
+      this.isResultActionLoading = false;
+      downloadAborted = true;
+    });
 
-    this.feedbackSessionsService.downloadSessionResults(
-      model.feedbackSession.courseId,
-      model.feedbackSession.feedbackSessionName,
-      Intent.INSTRUCTOR_RESULT,
-      true,
-      true,
-    ) .pipe(finalize(() => this.isResultActionLoading = false))
-      .subscribe((resp: string) => {
-        blob = new Blob([resp], { type: 'text/csv' });
-        saveAs(blob, filename);
-      }, (resp: ErrorMessageOutput) => {
-        this.statusMessageService.showErrorToast(resp.error.message);
-      });
+    this.feedbackQuestionsService.getFeedbackQuestions({
+      courseId: model.feedbackSession.courseId,
+      feedbackSessionName: model.feedbackSession.feedbackSessionName,
+      intent: Intent.INSTRUCTOR_RESULT,
+    }).subscribe((feedbackQuestions: FeedbackQuestions) => {
+      const questions: FeedbackQuestion[] = feedbackQuestions.questions;
```

Comment:
lets wrap this api calling part in a separate function e.g downloadSessionResult()